4. Stack/priority_tasks/PriorityTasksArnab.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * PriorityTasksArnab.java
+ * Given an array of priority values,we have to check if the value next is of higher priority or not.If yes, then print that value,else
```

Comment:
The problem says that we have to find the next higher priority element for a task, and if it doesn't exist, then it is the highest priority task. The array is circular in nature, so it will wrap up like:

1 --> 2 --> 3 --> 4 -->
|                                  |
|-----------------------|

So for the last element, search till (n - 2)th element. I hope this is clear.

Reply:
> The problem says that we have to find the next higher priority element for a task, and if it doesn't exist, then it is the highest priority task. The array is circular in nature, so it will wrap up like:
> 
> 1 --> 2 --> 3 --> 4 -->
> 
> So for the last element, search till (n - 2)th element. I hope this is clear.

Yaa yaa ,that I understood, Actually in the code I have checked uptill n-2 itself, but in the description I have written the entire array, I will correct it!